processor/src/main/java/de/hysky/skyblocker/processors/InitAnnotationProcessor.java

```diff
@@ -0,0 +1,70 @@
+package de.hysky.skyblocker.processors;
+
+import javax.annotation.processing.*;
+import javax.lang.model.SourceVersion;
+import javax.lang.model.element.*;
+import javax.tools.Diagnostic;
+import javax.tools.JavaFileObject;
+import java.io.IOException;
+import java.io.Writer;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+@SupportedAnnotationTypes(InitAnnotationProcessor.INIT)
+@SupportedSourceVersion(SourceVersion.RELEASE_21)
+public class InitAnnotationProcessor extends AbstractProcessor {
+	public static final String INIT = "de.hysky.skyblocker.annotations.Init";
+	private static final String GENERATED_CLASS_NAME = "de.hysky.skyblocker.Initializer";
+	private static final String GENERATED_METHOD_NAME = "init";
+
+	@Override
+	public boolean process(Set<? extends TypeElement> annotations, RoundEnvironment roundEnv) {
+		List<String> methodCalls = new ArrayList<>();
+
+		for (TypeElement annotation : annotations) {
+			for (Element element : roundEnv.getElementsAnnotatedWith(annotation)) {
+				if (!annotation.getQualifiedName().contentEquals(INIT)) continue;
+				Set<Modifier> modifiers = element.getModifiers();
+				if (element.getKind() == ElementKind.METHOD) {
+					if (!modifiers.contains(Modifier.PUBLIC) || !modifiers.contains(Modifier.STATIC)) {
```

Comment:
we should also ensure the method has no arguments

Reply:
Stale